### PR TITLE
[BUG] corrected default loss function to `CNNClassifier`

### PR DESCRIPTION
**Pull request by Vasudeva-bit**
#### Reference Issues/PRs

PR of sktime-dl [there](https://github.com/sktime/sktime-dl/pull/128) is migrated here.

#### What does this implement/fix? Explain your changes.

I have modified the default loss function to `CNNClassifier`, based on the conversations [here](https://github.com/sktime/sktime-dl/pull/128), from `mean_squared_error` to `categorical_crossentropy`.

#### Does your contribution introduce a new dependency? If yes, which one?

No.
#### What should a reviewer concentrate their feedback on?

If it is any better to approach this fix similar to handling `optimizer` parameter in the same classifier [here](https://github.com/sktime/sktime/blob/0efea1fcf240fad8f7775a41c6627781707add1f/sktime/classification/deep_learning/cnn.py#L151), I shall modify accordingly.

#### Did you add any tests for the change?

No

### CODE REVIEW

**Review by achieveordie — DISMISSED** [review]
LGTM, thanks!

**Comment by achieveordie** [comment]
Regarding your question, I don't believe we need any if statements since we already have a default.

What would be great though is to checkout whether other dl classifiers can be improved in the same manner.

**Comment by achieveordie** [comment]
Ah oh, please update the docstring as well :)

**Comment by fkiraly** [comment]
hm, question from my side, FYI @yarnabrina, @benHeid - does this need deprecation? Behaviour is changed, but MSE is a unusual choice for classification.

**Comment by yarnabrina** [comment]
> does this need deprecation?

In my opinion, yes. But considering unusual choice of loss, we can possibly do a shorter deprecation (1 minor) instead 2 as usual.

**Comment by achieveordie** [comment]
fwiw, I think it most probably was a mistake when copy-pasting it from `CNNRegressor` hence it does not warrant any deprecation.

**Comment by Vasudeva-bit** [comment]
> Also, is "sigmoid" a good choice for `categorical_crossentropy`, or `softmax` is a better choice?

I think `softmax` should be the default choice as it works better in case of multi-class classification and works just like `sigmoid` in case of binary classification. However, `sigmoid` is more appropriate choice in case of multi-label classification. Sorry, Shall I go ahead and change the default choice for activation?

**Comment by yarnabrina** [comment]
> I think softmax should be the default choice as it works better in case of multi-class classification and works just like sigmoid in case of binary classification.

Yes, my comment from this exact point of view. Multi-class is more common than multi-label, so I'll suggest to change the default loss. May be @achieveordie / @fkiraly  can share their opinions too?

File: sktime/classification/deep_learning/cnn.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class CNNClassifier(BaseDeepClassifier):
         Seed for random number generation.
     verbose         : boolean, default = False
         whether to output extra information
-    loss            : string, default="mean_squared_error"
+    loss            : string, default="categorical_crossentropy"
         fit parameter for the keras model
     metrics         : list of strings, default=["accuracy"],
     activation      : string or a tf callable, default="sigmoid"
```
Comment (yarnabrina) [inline sktime/classification/deep_learning/cnn.py:38]:
```suggestion
    activation      : string or a tf callable, default="softmax"
```

This should also be changed now.

**Review by yarnabrina — CHANGES_REQUESTED** [review]
LGTM otherwise, except for possible deprecation for at least one minor cycle.

**Review by yarnabrina — APPROVED** [review]
